src/java/org/apache/cassandra/cql3/ConstraintFunction.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.cql3;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.cassandra.io.util.DataInputPlus;
+import org.apache.cassandra.io.util.DataOutputPlus;
+import org.apache.cassandra.schema.TableMetadata;
+import org.apache.cassandra.tcm.serialization.Version;
+
+public class ConstraintFunction
+{
+    public final CqlConstraintFunctionExecutor executor;
+    public final List<ColumnIdentifier> arg;
+
+    public static final Serializer serializer = new Serializer();
+
+    public ConstraintFunction(CqlConstraintFunctionExecutor executor, List<ColumnIdentifier> arg)
+    {
+        this.executor = executor;
+        this.arg = arg;
+    }
+
+
+    public void checkConstraint(Operator relationType, String term, TableMetadata tableMetadata, Map<String, String> columnValues)
+    {
+        executor.checkConstraint(this.arg, relationType, term, tableMetadata, columnValues);
+    }
+
+    public void validateConstraint(Operator relationType, String term, TableMetadata tableMetadata)
+    {
+        executor.validate(this.arg, relationType, term, tableMetadata);
+    }
+
+    @Override
+    public String toString()
+    {
+        List<String> argsString = new ArrayList<>();
+        for (ColumnIdentifier columnIdentifier : arg)
+        {
+            argsString.add(columnIdentifier.toCQLString());
+        }
+        String args = String.join(", ", argsString);
+        return String.format("%s(%s)", executor.getName(), args);
+    }
```

Comment:
In addition to this, it also requires another method `toCqlString` to print as part of CQL statement literal output, e.g. when describing the schema.

Reply:
This is the method called from the `toCqlString` from places like TableMetadata. I don't think `toCqlString` specific method is needed at this level, as it never should be called directly I think? Or, do you think we need to provide to actual methods returning different? I can add the method returning `toString` to keep the pattern.